### PR TITLE
Feature/config gdev 138

### PR DESCRIPTION
**Pull request by KerstenBreuer**
This adds a python-based helper that takes config parameters from a config yaml, env variables, or file secret. It implements the pydantic-based functionality from the ghga_service_chassis_lib.

The basic usage is documented in:
- https://github.com/ghga-de/sandbox-ui/tree/feature/config-GDEV-138#configuration
- https://github.com/ghga-de/sandbox-ui/tree/feature/config-GDEV-138#building-and-serving-the

### CODE REVIEW

File: configure_build_serve/run.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            name not in IGNORE_PARAMS_FOR_REACT_APP and
+            value is not None
+        ):
+            env_var_name = f"REACT_APP_{name.upper()}"
+            os.environ[env_var_name] = str(value)
+            print(f"  - set {name} as {env_var_name}")
+
+
+def build():
+    """Builds a production read version of the web app"""
```
Comment (lkuchenb) [inline configure_build_serve/run.py:60]:
```suggestion
    """Builds a production ready version of the web app"""
```

File: configure_build_serve/run.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        "yarn", 
+        "--cwd",
+        str(ROOT_DIR),
+        "build"
+    ]
+    exit_code_build = Popen(cmd_build).wait()
+    
+    if exit_code_build != 0:
+        raise RuntimeError(
+            f"`yarn` ended with non-zero exit code: {exit_code_build}."
```
Comment (lkuchenb) [inline configure_build_serve/run.py:73]:
```suggestion
            f"`yarn` terminated with non-zero exit code: {exit_code_build}."
```

File: configure_build_serve/run.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        "yarn", 
+        "global",
+        "add",
+        "serve",
+    ]
+    exit_code_add_serve = Popen(cmd_add_serve).wait()
+    
+    if exit_code_add_serve != 0:
+        raise RuntimeError(
+            f"`yarn` ended with non-zero exit code: {exit_code_add_serve}."
```
Comment (lkuchenb) [inline configure_build_serve/run.py:91]:
```suggestion
            f"`yarn` terminated with non-zero exit code: {exit_code_add_serve}."
```

File: configure_build_serve/run.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    if exit_code_build != 0:
+        raise RuntimeError(
+            f"`yarn` ended with non-zero exit code: {exit_code_build}."
+        ) 
+
+
+def serve(config: Config):
+    """Serves a production ready build of the web app"""
+
+    simplelog("Making the `serve` package globally availabe:")
```
Comment (lkuchenb) [inline configure_build_serve/run.py:80]:
```suggestion
    simplelog("Making the `serve` package globally available:")
```

**Review by lkuchenb — APPROVED** [review]
The wrapper code looks good to me, minor remarks below (one of them before I realized the branch wasn't rebased, so I was looking at already merged commits).

One thought that crossed my mind is that react env variables that are set directly and are not overwritten by the config will silently influence the apps behavior. But I don't think that has any relevant consequences for us.

File: .devcontainer/test.sh [hunk 5]
```diff
@@ -0,0 +1,6 @@
+echo "deb http://http.us.debian.org/debian/ testing non-free contrib main" >> \
+    /etc/apt/sources.list
+apt update -y
+apt install -y python3.9 python3-pip
+update-alternatives --install /usr/bin/python python /usr/bin/python3.9 10
+update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.9 10
```
[the remark this answers is not in the document]
Comment (KerstenBreuer) [reply .devcontainer/test.sh:6]: Ups thats some debugging left-over

**Pull request merged**